app/src/main/java/net/squanchy/support/config/DialogLayoutParameters.java

```diff
@@ -0,0 +1,38 @@
+package net.squanchy.support.config;
+
+import android.app.Activity;
+import android.view.ViewGroup;
+import android.view.Window;
+
+public final class DialogLayoutParameters {
+
+    private final FormFactorChecker formFactorChecker;
+    private final int height;
+
+    public static DialogLayoutParameters fullHeight(Activity activity) {
+        return new DialogLayoutParameters(
+                new FormFactorChecker(activity),
+                ViewGroup.LayoutParams.MATCH_PARENT
+        );
+    }
+
+    public static DialogLayoutParameters wrapHeight(Activity activity) {
+        return new DialogLayoutParameters(
+                new FormFactorChecker(activity),
+                ViewGroup.LayoutParams.WRAP_CONTENT
+        );
+    }
+
+    private DialogLayoutParameters(FormFactorChecker formFactorChecker, int height) {
+        this.formFactorChecker = formFactorChecker;
+        this.height = height;
+    }
+
+    public void applyTo(Window window) {
+        if (formFactorChecker.isTablet()) {
+            window.setLayout(ViewGroup.LayoutParams.WRAP_CONTENT, height);
```

Comment:
Do you usually prefer getting the resources, then getting a dimension, to simply using a constant? The resource also has to be a hack since `match_parent` is `-1` but you cannot have a pure number as a dimension, so you have to declare it as `-1px`. Ditto for `wrap_content` which is `-2`.